services/motion/motion.go

```diff
@@ -186,8 +186,115 @@ type PlanWithStatus struct {
 }
 
 // A Service controls the flow of moving components.
+//
+// Move example:
+//
+//	// Assumes a gripper with the name "my_gripper" is configured on the machine
+//	gripperName := gripper.Named("my_gripper")
+//
+//	// Get the pose of the gripper
+//	myArmMotionPose, err := motionService.GetPose(context.Background(), gripperName, referenceframe.World, nil, nil)
+//
+//	log.Printf("Position of my_gripper from the motion service: %v", myArmMotionPose.Pose().Point())
+//	log.Printf("Orientation of my_gripper from the motion service: %v", myArmMotionPose.Pose().Orientation())
+//
+// MoveOnMap example:
+//
+//	// Assumes a base with the name "my_base" is configured on the machine
+//	myBaseResourceName := base.Named("my_base")
+//	mySLAMServiceResourceName := slam.Named("my_slam_service")
+//
+//	// Define a destination Pose
+//	myPose := spatialmath.NewPoseFromPoint(r3.Vector{Y: 10})
+//
+//	// Move the base component to the destination pose
+//	executionID, err := motionService.MoveOnMap(context.Background(), motion.MoveOnMapReq{
+//		ComponentName: myBaseResourceName,
+//		Destination:   myPose,
+//		SlamName:      mySLAMServiceResourceName,
+//	})
+//
+//	err = motion.PollHistoryUntilSuccessOrError(
+//		context.Background(),
+//		motionService,
+//		time.Duration(time.Second),
+//		motion.PlanHistoryReq{
+//			ComponentName: myBaseResourceName,
+//			ExecutionID:   executionID,
+//		},
+//	)
+//
+// MoveOnGlobe example:
+//
+//	// Assumes a base with the name "my_base" is configured on the machine
+//	// Get the resource names of the base and movement sensor
+//	myBaseResourceName := base.Named("myBase")
+//	myMvmntSensorResourceName := movementsensor.Named("my_movement_sensor")
+//
+//	// Define a destination Point at the GPS coordinates [0, 0]
+//	myDestination := geo.NewPoint(0, 0)
+//
+//	// Move the base component to the designated geographic location, as reported by the movement sensor
+//	executionID, err := motionService.MoveOnGlobe(context.Background(), motion.MoveOnGlobeReq{
```

Comment:
This is more of a general question for these docs. Are we not interested in checking for errors in code samples?

Reply:
Indeed. I don’t feel super strongly either but since it would be basically exactly the same on all, I think we decided it’s kind of like instantiating the component on every single one. That is, it’s also up to the user.